lib/screens/signup_page.dart

```diff
@@ -22,141 +21,226 @@ class _SignupPageState extends State<SignupPage> {
+        print("exception" + exception);
       });
     }
   }
 
   Widget _form(BuildContext context) {
-    return Form(
+    return Container(
+      child: Form(
         key: _formKey,
         child: Column(
-          crossAxisAlignment: CrossAxisAlignment.center,
           children: <Widget>[
+            SizedBox(height: 40.0),
             TextFormField(
               keyboardType: TextInputType.emailAddress,
-              decoration: const InputDecoration(
-                hintText: 'Email',
+              decoration: InputDecoration(
+                labelText: 'Email',
+                filled: true,
+                fillColor: Color(0xF6F6F6F6),
+                enabledBorder: OutlineInputBorder(
+                  borderSide: BorderSide(
+                    color: Colors.grey[200],
+                  ),
+                ),
+                prefixIcon: Icon(Icons.email),
               ),
               validator: (value) {
                 if (value.isEmpty) {
-                  return 'Email needed';
+                  return 'Email required.';
                 } else if (!_emailValid.hasMatch(value)) {
-                  return 'You must enter a valid email address';
+                  return 'Please enter a valid email address.';
                 }
                 _userEmail = value.trim();
                 return null;
               },
-              textAlign: TextAlign.center,
             ),
+            Visibility(
+              visible: exception == "" ? false : true,
+              child: Text(
+                exception,
+                style: TextStyle(color: Colors.red),
+              ),
+            ),
+            SizedBox(height: 20.0),
             TextFormField(
-              obscureText: true,
-              decoration: const InputDecoration(
-                hintText: 'Password',
+              obscureText: _isHiddenPassword,
+              decoration: InputDecoration(
+                labelText: 'Password',
+                filled: true,
+                fillColor: Color(0xF6F6F6F6),
+                enabledBorder: OutlineInputBorder(
+                  borderSide: BorderSide(
+                    color: Colors.grey[200],
+                  ),
+                ),
+                prefixIcon: Icon(Icons.lock),
+                suffixIcon: IconButton(
+                  onPressed: () {
+                    setState(() {
+                      _isHiddenPassword = !_isHiddenPassword;
+                    });
+                  },
+                  icon: Icon(
+                    _isHiddenPassword ? Icons.visibility : Icons.visibility_off,
+                  ),
+                ),
               ),
               validator: (value) {
                 if (value.isEmpty) {
-                  return 'Password Needed';
+                  return 'Password required.';
+                } else if (!_passwordValid.hasMatch(value)) {
+                  return 'Please enter a valid password.';
```

Comment:
I feel we should specify password requirements implicitly. 'Please enter a valid password.' is too vague.  Just add min char count, lower case, upper case, etc to the error message